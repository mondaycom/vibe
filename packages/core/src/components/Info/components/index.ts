export { default as InfoLink, type InfoLinkProps } from "./InfoLink/InfoLink";
export { default as InfoDialogContent } from "./InfoDialogContent/InfoDialogContent";
export type { InfoDialogContentProps } from "./InfoDialogContent/InfoDialogContent.types";
