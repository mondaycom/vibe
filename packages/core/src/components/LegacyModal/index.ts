export { default as Modal, type LegacyModalProps as ModalProps } from "./LegacyModal";
export {
  default as ModalHeader,
  type LegacyModalHeaderProps as ModalHeaderProps
} from "./LegacyModalHeader/LegacyModalHeader";
export {
  default as ModalContent,
  type LegacyModalContentProps as ModalContentProps
} from "./LegacyModalContent/LegacyModalContent";
export {
  default as ModalFooter,
  type LegacyModalFooterProps as ModalFooterProps
} from "./LegacyModalFooter/LegacyModalFooter";
export {
  default as ModalFooterButtons,
  type LegacyModalFooterButtonsProps as ModalFooterButtonsProps
} from "./LegacyModalFooter/LegacyModalFooterButtons/LegacyModalFooterButtons";

export * from "./LegacyModal.types";
