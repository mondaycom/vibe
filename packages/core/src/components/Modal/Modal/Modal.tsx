import React, { forwardRef, useCallback, useMemo, useRef, useState } from "react";
import cx from "classnames";
import { RemoveScroll } from "react-remove-scroll";
import FocusLock from "react-focus-lock";
import { motion, AnimatePresence } from "framer-motion";
import { getTestId } from "../../../tests/test-ids-utils";
import { ComponentDefaultTestId, ComponentVibeId } from "../../../tests/constants";
import styles from "./Modal.module.scss";
import { type ModalProps } from "./Modal.types";
import ModalTopActions from "../ModalTopActions/ModalTopActions";
import { getStyle } from "../../../helpers/typesciptCssModulesHelper";
import { camelCase } from "lodash-es";
import { ModalProvider } from "../context/ModalContext";
import { type ModalProviderValue } from "../context/ModalContext.types";
import { keyCodes } from "../../../constants";
import {
  modalAnimationAnchorPopVariants,
  modalAnimationCenterPopVariants,
  modalAnimationFullViewVariants,
  modalAnimationOverlayVariants
} from "../utils/animationVariants";
import { createPortal } from "react-dom";
import usePortalTarget from "../hooks/usePortalTarget/usePortalTarget";
import { LayerProvider } from "../../LayerProvider";
import useMergeRef from "../../../hooks/useMergeRef";

// @ts-expect-error This is a precaution to support all possible module systems (ESM/CJS)
const FocusLockComponent = (FocusLock.default || FocusLock) as typeof FocusLock;

const Modal = forwardRef(
  (
    {
      id,
      show,
      size = "medium",
      renderHeaderAction,
      closeButtonTheme,
      closeButtonAriaLabel,
      onClose = () => {},
      autoFocus = true,
      onFocusAttempt,
      anchorElementRef,
      alertModal,
      container = document.body,
      children,
      style,
      zIndex,
      className,
      "data-testid": dataTestId,
      "aria-labelledby": ariaLabelledby,
      "aria-describedby": ariaDescribedby
    }: ModalProps,
    ref: React.ForwardedRef<HTMLDivElement>
  ) => {
    const portalTargetElement = usePortalTarget(container);

    const modalRef = useRef<HTMLDivElement>(null);
    const modalMergedRef = useMergeRef<HTMLDivElement>(ref, modalRef);
    const containerRef = useRef<HTMLDivElement>(null);

    const [titleId, setTitleId] = useState<string>();
    const [descriptionId, setDescriptionId] = useState<string>();

    const setTitleIdCallback = useCallback(
      (newId: string) => {
        if (ariaLabelledby) return;
        setTitleId(newId);
      },
      [ariaLabelledby]
    );
    const setDescriptionIdCallback = useCallback(
      (newId: string) => {
        if (ariaDescribedby) return;
        setDescriptionId(newId);
      },
      [ariaDescribedby]
    );

    const contextValue = useMemo<ModalProviderValue>(
      () => ({
        modalId: id,
        setTitleId: setTitleIdCallback,
        setDescriptionId: setDescriptionIdCallback,
        autoFocus
      }),
      [id, setTitleIdCallback, setDescriptionIdCallback, autoFocus]
    );

    const onBackdropClick = useCallback<React.MouseEventHandler<HTMLDivElement>>(
      e => {
        if (!show || alertModal) return;
        onClose(e);
      },
      [show, alertModal, onClose]
    );

    const onModalKeyDown = useCallback<React.KeyboardEventHandler<HTMLDivElement>>(
      e => {
        if (e.key !== keyCodes.ESCAPE || !show || alertModal) return;
        onClose(e);
      },
      [alertModal, onClose, show]
    );

    const modalAnimationVariants =
      size === "full-view"
        ? modalAnimationFullViewVariants
        : anchorElementRef?.current
        ? modalAnimationAnchorPopVariants
        : modalAnimationCenterPopVariants;

    const zIndexStyle = zIndex ? ({ "--monday-modal-z-index": zIndex } as React.CSSProperties) : {};

    const handleFocusLockWhiteList = useCallback(
      (nextFocusedElement?: HTMLElement) => {
        if (!onFocusAttempt) return true;

        const outcome = onFocusAttempt(nextFocusedElement);

        if (outcome === true) return true;

        if (outcome instanceof HTMLElement) {
          outcome.focus();
          return false;
        }

        return false;
      },
      [onFocusAttempt]
    );

    return (
      <AnimatePresence>
        {show && (
          <LayerProvider layerRef={containerRef}>
            <ModalProvider value={contextValue}>
              {createPortal(
                <FocusLockComponent returnFocus autoFocus={autoFocus} whiteList={handleFocusLockWhiteList}>
                  <div ref={containerRef} className={styles.container} style={zIndexStyle}>
                    <motion.div
                      variants={modalAnimationOverlayVariants}
                      initial="initial"
                      animate="enter"
                      exit="exit"
                      data-testid={getTestId(ComponentDefaultTestId.MODAL_NEXT_OVERLAY, id)}
                      className={styles.overlay}
                      onClick={onBackdropClick}
                      aria-hidden
                    />
                    <RemoveScroll forwardProps ref={modalMergedRef}>
                      <motion.div
                        variants={modalAnimationVariants}
                        initial="exit"
                        animate="enter"
                        exit="exit"
                        custom={anchorElementRef}
                        className={cx(
                          styles.modal,
                          getStyle(styles, camelCase("size-" + size)),
                          { [styles.withHeaderAction]: !!renderHeaderAction },
                          className
                        )}
                        id={id}
                        data-testid={dataTestId || getTestId(ComponentDefaultTestId.MODAL_NEXT, id)}
                        data-vibe={ComponentVibeId.MODAL}
                        role="dialog"
                        aria-modal
                        aria-labelledby={ariaLabelledby || titleId}
                        aria-describedby={ariaDescribedby || descriptionId}
                        style={style}
                        onKeyDown={onModalKeyDown}
                        tabIndex={-1}
                      >
                        {children}
                        <ModalTopActions
                          renderAction={renderHeaderAction}
                          theme={closeButtonTheme}
                          closeButtonAriaLabel={closeButtonAriaLabel}
                          onClose={onClose}
                        />
                      </motion.div>
                    </RemoveScroll>
                  </div>
                </FocusLockComponent>,
                portalTargetElement
              )}
            </ModalProvider>
          </LayerProvider>
        )}
      </AnimatePresence>
    );
  }
);

export default Modal;
