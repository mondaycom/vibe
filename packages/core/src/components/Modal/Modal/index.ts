export { default as Modal } from "./Modal";
export { type ModalProps } from "./Modal.types";
