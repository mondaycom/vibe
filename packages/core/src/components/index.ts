export * from "./Accordion";
export * from "./AlertBanner";
export * from "./AttentionBox";
export * from "./Avatar";
export * from "./AvatarGroup";
export * from "./Badge";
export * from "./Box";
export * from "./BreadcrumbsBar";
export * from "./Button";
export * from "./ButtonGroup";
export * from "./Checkbox";
export * from "./Chips";
export * from "./Clickable";
export * from "./ColorPicker";
export * from "./Combobox";
export * from "./Counter";
export * from "./DatePicker";
export * from "./Dialog";
export * from "./DialogContentContainer";
export * from "./Divider";
export * from "./Dropdown";
export * from "./EditableHeading";
export * from "./EditableText";
export * from "./EmptyState";
export * from "./ExpandCollapse";
export * from "./Flex";
export * from "./FormattedNumber";
export * from "./GridKeyboardNavigationContext";
export * from "./Heading";
export * from "./HiddenText";
export * from "./Icon";
export * from "./IconButton";
export * from "./Label";
export * from "./LayerProvider";
export * from "./Link";
export * from "./List";
export * from "./ListItem";
export * from "./ListItemAvatar";
export * from "./ListItemIcon";
export * from "./ListTitle";
export * from "./Loader";
export * from "./Menu";
export * from "./MenuButton";
export * from "./LegacyModal";
export * from "./MultiStepIndicator";
export * from "./ProgressBars";
export * from "./RadioButton";
export * from "./Search";
export * from "./Skeleton";
export * from "./Slider";
export * from "./SplitButton";
export * from "./Steps";
export * from "./Table";
export * from "./Tabs";
export * from "./Text";
export * from "./TextArea";
export * from "./TextField";
export * from "./TextWithHighlight";
export * from "./ThemeProvider";
export * from "./Tipseen";
export * from "./Toast";
export * from "./Toggle";
export * from "./Tooltip";
export * from "./TransitionView";
export * from "./VirtualizedGrid";
export * from "./VirtualizedList";

export type { TypographyColor, TypographyAlign } from "./Typography";

export { default as ColorUtils } from "../utils/colors-utils";
