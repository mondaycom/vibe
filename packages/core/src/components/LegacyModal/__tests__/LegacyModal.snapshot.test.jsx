import { describe, it, expect } from "vitest";
import React from "react";
import Modal from "../LegacyModal";
import { ModalContent, ModalFooter, ModalHeader } from "../../../components";
import { render } from "@testing-library/react";

const baseProps = {
  id: "modal-id",
  show: true,
  title: "Modal Heading",
  onClose: () => {}
};

const content = <p>content</p>;
const footerContent = <p>footer</p>;

describe("Modal renders correctly", () => {
  it("with base props", () => {
    const { baseElement } = render(<Modal {...baseProps}>{content}</Modal>);
    expect(baseElement).toMatchSnapshot();
  });

  it("with Header", () => {
    const { baseElement } = render(
      <Modal {...baseProps}>
        <ModalHeader title="ModalHeader Heading" />
        {content}
      </Modal>
    );
    expect(baseElement).toMatchSnapshot();
  });

  it("with Footer", () => {
    const { baseElement } = render(
      <Modal {...baseProps}>
        {content}
        <ModalFooter>{footerContent}</ModalFooter>
      </Modal>
    );
    expect(baseElement).toMatchSnapshot();
  });

  it("with description", () => {
    const { baseElement } = render(
      <Modal {...baseProps} description="description">
        {content}
      </Modal>
    );
    expect(baseElement).toMatchSnapshot();
  });

  it("with alertDialog", () => {
    const { baseElement } = render(
      <Modal {...baseProps} isAlertDialog={true}>
        {content}
      </Modal>
    );
    expect(baseElement).toMatchSnapshot();
  });

  it("with closeButtonAriaLabel", () => {
    const { baseElement } = render(
      <Modal {...baseProps} closeButtonAriaLabel="closeButtonAriaLabel">
        {content}
      </Modal>
    );
    expect(baseElement).toMatchSnapshot();
  });

  it("with full width", () => {
    const { baseElement } = render(
      <Modal {...baseProps} width="full-width">
        {content}
      </Modal>
    );
    expect(baseElement).toMatchSnapshot();
  });

  it("with custom width", () => {
    const { baseElement } = render(
      <Modal {...baseProps} width="720px">
        {content}
      </Modal>
    );
    expect(baseElement).toMatchSnapshot();
  });

  it("with class names", () => {
    const { baseElement } = render(
      <Modal {...baseProps} classNames={{ container: "container", overlay: "overlay", modal: "modal" }}>
        {content}
      </Modal>
    );
    expect(baseElement).toMatchSnapshot();
  });

  it("with all parts", () => {
    const { baseElement } = render(
      <Modal {...baseProps}>
        <ModalHeader title="ModalHeader Heading" />
        <ModalContent>{content}</ModalContent>
        <ModalFooter>{footerContent}</ModalFooter>
      </Modal>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
