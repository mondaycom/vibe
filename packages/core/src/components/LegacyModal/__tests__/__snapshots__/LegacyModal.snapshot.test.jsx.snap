// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Modal renders correctly > with Footer 1`] = `
<body
  style="overflow: hidden;"
>
  <div />
  <div
    aria-labelledby="modal-id-title"
    aria-modal="true"
    class="container"
    data-testid="monday-dialog-container_modal-id"
    id="modal-id"
    role="dialog"
    style="--monday-modal-z-index: 10000;"
    tabindex="-1"
  >
    <div
      class="overlay"
      data-testid="monday-modal-overlay"
    />
    <div
      class="dialog default"
      role="document"
    >
      <div
        class="container"
        data-testid="modal-header_modal-id-title"
      >
        <div
          class="container directionRow justifyStart alignStart"
          style="gap: var(--spacing-small);"
        >
          <h1
            class="typography primary start multiLineEllipsis heading h1Normal"
            data-testid="text_modal-id-title"
            id="modal-id-title"
            style="--text-clamp-lines: 2;"
          >
            Modal Heading
          </h1>
        </div>
        
        <div
          class="closeButton"
        >
          <span
            class="referenceWrapper"
          >
            <button
              aria-busy="false"
              aria-disabled="false"
              aria-label="Close"
              class="button sizeMedium kindTertiary colorPrimary noSidePadding"
              data-testid="icon-button"
              data-vibe="Button"
              style="justify-content: center; align-items: center; padding: 0px; width: 32px; height: 32px;"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="icon noFocusStyle"
                data-testid="icon"
                data-vibe="Icon"
                fill="currentColor"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M6.331 5.27a.75.75 0 0 0-1.06 1.06L8.94 10l-3.67 3.668a.75.75 0 1 0 1.06 1.06L10 11.06l3.668 3.669a.75.75 0 0 0 1.06-1.06l-3.668-3.67 3.67-3.669a.75.75 0 0 0-1.061-1.06L10 8.939l-3.669-3.67Z"
                />
              </svg>
            </button>
          </span>
        </div>
      </div>
      <div
        class="typography primary start text text1Normal container"
        data-testid="modal-content"
      >
        <p>
          content
        </p>
      </div>
      <div
        class="container"
      >
        <p>
          footer
        </p>
      </div>
    </div>
  </div>
</body>
`;

exports[`Modal renders correctly > with Header 1`] = `
<body
  style="overflow: hidden;"
>
  <div />
  <div
    aria-labelledby="modal-id-title"
    aria-modal="true"
    class="container"
    data-testid="monday-dialog-container_modal-id"
    id="modal-id"
    role="dialog"
    style="--monday-modal-z-index: 10000;"
    tabindex="-1"
  >
    <div
      class="overlay"
      data-testid="monday-modal-overlay"
    />
    <div
      class="dialog default"
      role="document"
    >
      <div
        class="container"
        data-testid="modal-header_modal-id-title"
      >
        <div
          class="container directionRow justifyStart alignStart"
          style="gap: var(--spacing-small);"
        >
          <h1
            class="typography primary start multiLineEllipsis heading h1Normal"
            data-testid="text_modal-id-title"
            id="modal-id-title"
            style="--text-clamp-lines: 2;"
          >
            ModalHeader Heading
          </h1>
        </div>
        
        <div
          class="closeButton"
        >
          <span
            class="referenceWrapper"
          >
            <button
              aria-busy="false"
              aria-disabled="false"
              aria-label="close"
              class="button sizeMedium kindTertiary colorPrimary noSidePadding"
              data-testid="icon-button"
              data-vibe="Button"
              style="justify-content: center; align-items: center; padding: 0px; width: 32px; height: 32px;"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="icon noFocusStyle"
                data-testid="icon"
                data-vibe="Icon"
                fill="currentColor"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M6.331 5.27a.75.75 0 0 0-1.06 1.06L8.94 10l-3.67 3.668a.75.75 0 1 0 1.06 1.06L10 11.06l3.668 3.669a.75.75 0 0 0 1.06-1.06l-3.668-3.67 3.67-3.669a.75.75 0 0 0-1.061-1.06L10 8.939l-3.669-3.67Z"
                />
              </svg>
            </button>
          </span>
        </div>
      </div>
      <div
        class="typography primary start text text1Normal container"
        data-testid="modal-content"
      >
        <p>
          content
        </p>
      </div>
    </div>
  </div>
</body>
`;

exports[`Modal renders correctly > with alertDialog 1`] = `
<body
  style="overflow: hidden;"
>
  <div />
  <div
    aria-labelledby="modal-id-title"
    aria-modal="true"
    class="container"
    data-testid="monday-dialog-container_modal-id"
    id="modal-id"
    role="dialog"
    style="--monday-modal-z-index: 10000;"
    tabindex="-1"
  >
    <div
      class="overlay"
      data-testid="monday-modal-overlay"
    />
    <div
      class="dialog default"
      role="document"
    >
      <div
        class="container"
        data-testid="modal-header_modal-id-title"
      >
        <div
          class="container directionRow justifyStart alignStart"
          style="gap: var(--spacing-small);"
        >
          <h1
            class="typography primary start multiLineEllipsis heading h1Normal"
            data-testid="text_modal-id-title"
            id="modal-id-title"
            style="--text-clamp-lines: 2;"
          >
            Modal Heading
          </h1>
        </div>
        
        <div
          class="closeButton"
        >
          <span
            class="referenceWrapper"
          >
            <button
              aria-busy="false"
              aria-disabled="false"
              aria-label="Close"
              class="button sizeMedium kindTertiary colorPrimary noSidePadding"
              data-testid="icon-button"
              data-vibe="Button"
              style="justify-content: center; align-items: center; padding: 0px; width: 32px; height: 32px;"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="icon noFocusStyle"
                data-testid="icon"
                data-vibe="Icon"
                fill="currentColor"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M6.331 5.27a.75.75 0 0 0-1.06 1.06L8.94 10l-3.67 3.668a.75.75 0 1 0 1.06 1.06L10 11.06l3.668 3.669a.75.75 0 0 0 1.06-1.06l-3.668-3.67 3.67-3.669a.75.75 0 0 0-1.061-1.06L10 8.939l-3.669-3.67Z"
                />
              </svg>
            </button>
          </span>
        </div>
      </div>
      <div
        class="typography primary start text text1Normal container"
        data-testid="modal-content"
      >
        <p>
          content
        </p>
      </div>
    </div>
  </div>
</body>
`;

exports[`Modal renders correctly > with all parts 1`] = `
<body
  style="overflow: hidden;"
>
  <div />
  <div
    aria-labelledby="modal-id-title"
    aria-modal="true"
    class="container"
    data-testid="monday-dialog-container_modal-id"
    id="modal-id"
    role="dialog"
    style="--monday-modal-z-index: 10000;"
    tabindex="-1"
  >
    <div
      class="overlay"
      data-testid="monday-modal-overlay"
    />
    <div
      class="dialog default"
      role="document"
    >
      <div
        class="container"
        data-testid="modal-header_modal-id-title"
      >
        <div
          class="container directionRow justifyStart alignStart"
          style="gap: var(--spacing-small);"
        >
          <h1
            class="typography primary start multiLineEllipsis heading h1Normal"
            data-testid="text_modal-id-title"
            id="modal-id-title"
            style="--text-clamp-lines: 2;"
          >
            ModalHeader Heading
          </h1>
        </div>
        
        <div
          class="closeButton"
        >
          <span
            class="referenceWrapper"
          >
            <button
              aria-busy="false"
              aria-disabled="false"
              aria-label="close"
              class="button sizeMedium kindTertiary colorPrimary noSidePadding"
              data-testid="icon-button"
              data-vibe="Button"
              style="justify-content: center; align-items: center; padding: 0px; width: 32px; height: 32px;"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="icon noFocusStyle"
                data-testid="icon"
                data-vibe="Icon"
                fill="currentColor"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M6.331 5.27a.75.75 0 0 0-1.06 1.06L8.94 10l-3.67 3.668a.75.75 0 1 0 1.06 1.06L10 11.06l3.668 3.669a.75.75 0 0 0 1.06-1.06l-3.668-3.67 3.67-3.669a.75.75 0 0 0-1.061-1.06L10 8.939l-3.669-3.67Z"
                />
              </svg>
            </button>
          </span>
        </div>
      </div>
      <div
        class="typography primary start text text1Normal container"
        data-testid="modal-content"
      >
        <p>
          content
        </p>
      </div>
      <div
        class="container"
      >
        <p>
          footer
        </p>
      </div>
    </div>
  </div>
</body>
`;

exports[`Modal renders correctly > with base props 1`] = `
<body
  style="overflow: hidden;"
>
  <div />
  <div
    aria-labelledby="modal-id-title"
    aria-modal="true"
    class="container"
    data-testid="monday-dialog-container_modal-id"
    id="modal-id"
    role="dialog"
    style="--monday-modal-z-index: 10000;"
    tabindex="-1"
  >
    <div
      class="overlay"
      data-testid="monday-modal-overlay"
    />
    <div
      class="dialog default"
      role="document"
    >
      <div
        class="container"
        data-testid="modal-header_modal-id-title"
      >
        <div
          class="container directionRow justifyStart alignStart"
          style="gap: var(--spacing-small);"
        >
          <h1
            class="typography primary start multiLineEllipsis heading h1Normal"
            data-testid="text_modal-id-title"
            id="modal-id-title"
            style="--text-clamp-lines: 2;"
          >
            Modal Heading
          </h1>
        </div>
        
        <div
          class="closeButton"
        >
          <span
            class="referenceWrapper"
          >
            <button
              aria-busy="false"
              aria-disabled="false"
              aria-label="Close"
              class="button sizeMedium kindTertiary colorPrimary noSidePadding"
              data-testid="icon-button"
              data-vibe="Button"
              style="justify-content: center; align-items: center; padding: 0px; width: 32px; height: 32px;"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="icon noFocusStyle"
                data-testid="icon"
                data-vibe="Icon"
                fill="currentColor"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M6.331 5.27a.75.75 0 0 0-1.06 1.06L8.94 10l-3.67 3.668a.75.75 0 1 0 1.06 1.06L10 11.06l3.668 3.669a.75.75 0 0 0 1.06-1.06l-3.668-3.67 3.67-3.669a.75.75 0 0 0-1.061-1.06L10 8.939l-3.669-3.67Z"
                />
              </svg>
            </button>
          </span>
        </div>
      </div>
      <div
        class="typography primary start text text1Normal container"
        data-testid="modal-content"
      >
        <p>
          content
        </p>
      </div>
    </div>
  </div>
</body>
`;

exports[`Modal renders correctly > with class names 1`] = `
<body
  style="overflow: hidden;"
>
  <div />
  <div
    aria-labelledby="modal-id-title"
    aria-modal="true"
    class="container container"
    data-testid="monday-dialog-container_modal-id"
    id="modal-id"
    role="dialog"
    style="--monday-modal-z-index: 10000;"
    tabindex="-1"
  >
    <div
      class="overlay overlay"
      data-testid="monday-modal-overlay"
    />
    <div
      class="dialog modal default"
      role="document"
    >
      <div
        class="container"
        data-testid="modal-header_modal-id-title"
      >
        <div
          class="container directionRow justifyStart alignStart"
          style="gap: var(--spacing-small);"
        >
          <h1
            class="typography primary start multiLineEllipsis heading h1Normal"
            data-testid="text_modal-id-title"
            id="modal-id-title"
            style="--text-clamp-lines: 2;"
          >
            Modal Heading
          </h1>
        </div>
        
        <div
          class="closeButton"
        >
          <span
            class="referenceWrapper"
          >
            <button
              aria-busy="false"
              aria-disabled="false"
              aria-label="Close"
              class="button sizeMedium kindTertiary colorPrimary noSidePadding"
              data-testid="icon-button"
              data-vibe="Button"
              style="justify-content: center; align-items: center; padding: 0px; width: 32px; height: 32px;"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="icon noFocusStyle"
                data-testid="icon"
                data-vibe="Icon"
                fill="currentColor"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M6.331 5.27a.75.75 0 0 0-1.06 1.06L8.94 10l-3.67 3.668a.75.75 0 1 0 1.06 1.06L10 11.06l3.668 3.669a.75.75 0 0 0 1.06-1.06l-3.668-3.67 3.67-3.669a.75.75 0 0 0-1.061-1.06L10 8.939l-3.669-3.67Z"
                />
              </svg>
            </button>
          </span>
        </div>
      </div>
      <div
        class="typography primary start text text1Normal container"
        data-testid="modal-content"
      >
        <p>
          content
        </p>
      </div>
    </div>
  </div>
</body>
`;

exports[`Modal renders correctly > with closeButtonAriaLabel 1`] = `
<body
  style="overflow: hidden;"
>
  <div />
  <div
    aria-labelledby="modal-id-title"
    aria-modal="true"
    class="container"
    data-testid="monday-dialog-container_modal-id"
    id="modal-id"
    role="dialog"
    style="--monday-modal-z-index: 10000;"
    tabindex="-1"
  >
    <div
      class="overlay"
      data-testid="monday-modal-overlay"
    />
    <div
      class="dialog default"
      role="document"
    >
      <div
        class="container"
        data-testid="modal-header_modal-id-title"
      >
        <div
          class="container directionRow justifyStart alignStart"
          style="gap: var(--spacing-small);"
        >
          <h1
            class="typography primary start multiLineEllipsis heading h1Normal"
            data-testid="text_modal-id-title"
            id="modal-id-title"
            style="--text-clamp-lines: 2;"
          >
            Modal Heading
          </h1>
        </div>
        
        <div
          class="closeButton"
        >
          <span
            class="referenceWrapper"
          >
            <button
              aria-busy="false"
              aria-disabled="false"
              aria-label="closeButtonAriaLabel"
              class="button sizeMedium kindTertiary colorPrimary noSidePadding"
              data-testid="icon-button"
              data-vibe="Button"
              style="justify-content: center; align-items: center; padding: 0px; width: 32px; height: 32px;"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="icon noFocusStyle"
                data-testid="icon"
                data-vibe="Icon"
                fill="currentColor"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M6.331 5.27a.75.75 0 0 0-1.06 1.06L8.94 10l-3.67 3.668a.75.75 0 1 0 1.06 1.06L10 11.06l3.668 3.669a.75.75 0 0 0 1.06-1.06l-3.668-3.67 3.67-3.669a.75.75 0 0 0-1.061-1.06L10 8.939l-3.669-3.67Z"
                />
              </svg>
            </button>
          </span>
        </div>
      </div>
      <div
        class="typography primary start text text1Normal container"
        data-testid="modal-content"
      >
        <p>
          content
        </p>
      </div>
    </div>
  </div>
</body>
`;

exports[`Modal renders correctly > with custom width 1`] = `
<body
  style="overflow: hidden;"
>
  <div />
  <div
    aria-labelledby="modal-id-title"
    aria-modal="true"
    class="container"
    data-testid="monday-dialog-container_modal-id"
    id="modal-id"
    role="dialog"
    style="--monday-modal-z-index: 10000;"
    tabindex="-1"
  >
    <div
      class="overlay"
      data-testid="monday-modal-overlay"
    />
    <div
      class="dialog"
      role="document"
      style="width: 720px;"
    >
      <div
        class="container"
        data-testid="modal-header_modal-id-title"
      >
        <div
          class="container directionRow justifyStart alignStart"
          style="gap: var(--spacing-small);"
        >
          <h1
            class="typography primary start multiLineEllipsis heading h1Normal"
            data-testid="text_modal-id-title"
            id="modal-id-title"
            style="--text-clamp-lines: 2;"
          >
            Modal Heading
          </h1>
        </div>
        
        <div
          class="closeButton"
        >
          <span
            class="referenceWrapper"
          >
            <button
              aria-busy="false"
              aria-disabled="false"
              aria-label="Close"
              class="button sizeMedium kindTertiary colorPrimary noSidePadding"
              data-testid="icon-button"
              data-vibe="Button"
              style="justify-content: center; align-items: center; padding: 0px; width: 32px; height: 32px;"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="icon noFocusStyle"
                data-testid="icon"
                data-vibe="Icon"
                fill="currentColor"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M6.331 5.27a.75.75 0 0 0-1.06 1.06L8.94 10l-3.67 3.668a.75.75 0 1 0 1.06 1.06L10 11.06l3.668 3.669a.75.75 0 0 0 1.06-1.06l-3.668-3.67 3.67-3.669a.75.75 0 0 0-1.061-1.06L10 8.939l-3.669-3.67Z"
                />
              </svg>
            </button>
          </span>
        </div>
      </div>
      <div
        class="typography primary start text text1Normal container"
        data-testid="modal-content"
      >
        <p>
          content
        </p>
      </div>
    </div>
  </div>
</body>
`;

exports[`Modal renders correctly > with description 1`] = `
<body
  style="overflow: hidden;"
>
  <div />
  <div
    aria-labelledby="modal-id-title"
    aria-modal="true"
    class="container"
    data-testid="monday-dialog-container_modal-id"
    id="modal-id"
    role="dialog"
    style="--monday-modal-z-index: 10000;"
    tabindex="-1"
  >
    <div
      class="overlay"
      data-testid="monday-modal-overlay"
    />
    <div
      class="dialog default"
      role="document"
    >
      <div
        class="container"
        data-testid="modal-header_modal-id-title"
      >
        <div
          class="container directionRow justifyStart alignStart"
          style="gap: var(--spacing-small);"
        >
          <h1
            class="typography primary start multiLineEllipsis heading h1Normal"
            data-testid="text_modal-id-title"
            id="modal-id-title"
            style="--text-clamp-lines: 2;"
          >
            Modal Heading
          </h1>
        </div>
        <div
          class="typography primary start multiLineEllipsis text text2Normal description"
          data-testid="text"
          style="--text-clamp-lines: 2;"
        >
          description
        </div>
        <div
          class="closeButton"
        >
          <span
            class="referenceWrapper"
          >
            <button
              aria-busy="false"
              aria-disabled="false"
              aria-label="Close"
              class="button sizeMedium kindTertiary colorPrimary noSidePadding"
              data-testid="icon-button"
              data-vibe="Button"
              style="justify-content: center; align-items: center; padding: 0px; width: 32px; height: 32px;"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="icon noFocusStyle"
                data-testid="icon"
                data-vibe="Icon"
                fill="currentColor"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M6.331 5.27a.75.75 0 0 0-1.06 1.06L8.94 10l-3.67 3.668a.75.75 0 1 0 1.06 1.06L10 11.06l3.668 3.669a.75.75 0 0 0 1.06-1.06l-3.668-3.67 3.67-3.669a.75.75 0 0 0-1.061-1.06L10 8.939l-3.669-3.67Z"
                />
              </svg>
            </button>
          </span>
        </div>
      </div>
      <div
        class="typography primary start text text1Normal container"
        data-testid="modal-content"
      >
        <p>
          content
        </p>
      </div>
    </div>
  </div>
</body>
`;

exports[`Modal renders correctly > with full width 1`] = `
<body
  style="overflow: hidden;"
>
  <div />
  <div
    aria-labelledby="modal-id-title"
    aria-modal="true"
    class="container"
    data-testid="monday-dialog-container_modal-id"
    id="modal-id"
    role="dialog"
    style="--monday-modal-z-index: 10000;"
    tabindex="-1"
  >
    <div
      class="overlay"
      data-testid="monday-modal-overlay"
    />
    <div
      class="dialog full"
      role="document"
    >
      <div
        class="container"
        data-testid="modal-header_modal-id-title"
      >
        <div
          class="container directionRow justifyStart alignStart"
          style="gap: var(--spacing-small);"
        >
          <h1
            class="typography primary start multiLineEllipsis heading h1Normal"
            data-testid="text_modal-id-title"
            id="modal-id-title"
            style="--text-clamp-lines: 2;"
          >
            Modal Heading
          </h1>
        </div>
        
        <div
          class="closeButton"
        >
          <span
            class="referenceWrapper"
          >
            <button
              aria-busy="false"
              aria-disabled="false"
              aria-label="Close"
              class="button sizeMedium kindTertiary colorPrimary noSidePadding"
              data-testid="icon-button"
              data-vibe="Button"
              style="justify-content: center; align-items: center; padding: 0px; width: 32px; height: 32px;"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="icon noFocusStyle"
                data-testid="icon"
                data-vibe="Icon"
                fill="currentColor"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M6.331 5.27a.75.75 0 0 0-1.06 1.06L8.94 10l-3.67 3.668a.75.75 0 1 0 1.06 1.06L10 11.06l3.668 3.669a.75.75 0 0 0 1.06-1.06l-3.668-3.67 3.67-3.669a.75.75 0 0 0-1.061-1.06L10 8.939l-3.669-3.67Z"
                />
              </svg>
            </button>
          </span>
        </div>
      </div>
      <div
        class="typography primary start text text1Normal container"
        data-testid="modal-content"
      >
        <p>
          content
        </p>
      </div>
    </div>
  </div>
</body>
`;
